unit-tests/src/test/scala/java/lang/StringSuite.scala

```diff
@@ -255,10 +255,37 @@ object StringSuite extends tests.Suite {
     splitTest("ab", splitExpr = Some("(ab)"))
   }
 
-  test("getBytes") {
-    val b = new Array[scala.Byte](4)
-    "This is a test".getBytes(10, 14, b, 0)
-    assert(new String(b) equals "test")
+  test("getBytes(\"UTF-8\")") {
```

Comment:
It doesn't seem like these changes should be part of this PR.

Reply:
I like the way you think of things...

Reading your comment, it looked like a space shot to me also. I could not reverse engineer it to
provide a reason, so I spent some hours re-creating the original conditions:
```
[error] /Devo_1/PR_build_warnings_I1748_V2/unit-tests/src/test/scala/java/lang/StringSuite.scala:260:22: method getBytes in class String is deprecated: see corresponding Javadoc for more information.
[error]     "This is a test".getBytes(10, 14, b, 0)
[error]                      ^
[error] one error found
[error] (tests / Test / compileIncremental) Compilation failed
[error] Total time: 49 s, completed May 5, 2020 3:56:41 PM
```

The code of this PR gets rid of the deprecated overload of getBytes() and then
provides a reasonably strong test, including Unicode, for the current overload.

Reply:
Ah, I see what's happening.

Now that's a bit tricky, because the changes mean that we are not testing `getBytes(Array[Byte])` anymore. Even if it is deprecated, we must keep testing it. That shows that we have to live with deprecations *in the test suite*. This is inevitable.

Would adding `-deprecation:false` in the test suite project solve this? (like you did in the `scalalib` project)

Reply:
I rolled back the deprecation check in the test sub-project. That puts that sub-project at
"no worse than today"and allows the main benefit of this PR to advance.

Fate willing, I will file two PRs. One to add the UTF-8 test. As long as it is written and
working, it is worthwhile to capture the currently recommended case.

The second PR would add deprecation checking back in.  I need to sandbox 
some ideas from Scala 2.13.2 and the silencer plugin.

Reply:
It might be nice if you experimented with `sbt-tpolecat` if you find the time. It is the most strict but also is designed for cross project so if we release a 2.12/2.13 and eventually Scala 3/Dotty it should take the options in account as they are changed/added/removed from version to version.

Reply:
@ekrich Thank you, as always, for the on-point suggestion.  You could open an Issue for
 discussion and as a forget-me-not.

At the time I did this PR, I had looked at tpolecat's recommendation list and the date on
that list.  I chose the options I understood, mostly from having used them.  I also wanted
to chose options which did not disrupt the world too much. I left that for my no-unused-imports
PR :grimacing:  

Based on your recommendation, I just did a quick sweep of sbt-tpolecat. A topic for
a separate Issue.  Selected option, such a "partial-unification" and such look promising
but require a menagerie of plugins.   Out of scope for this PR, I think but worth discussing
in another venue.